### PR TITLE
feat: [DHIS2-10218] Stages and Events

### PR DESCRIPTION
**Pull request by huocha**

### CODE REVIEW

File: src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/Stage.component.js [hunk 1]
```diff
@@ -14,13 +17,35 @@ const styles = {
...
+                title={name}
+                icon={icon}
+                description={stageForm.description ?? ''}
+                events={events}
+            />}
+            onOpen={useCallback(() => setOpenStatus(true), [setOpenStatus])}
+            onClose={useCallback(() => setOpenStatus(false), [setOpenStatus])}
+            open={open}
+        >
+            {events.length > 0 && <StageDetail
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/Stage.component.js:41]: Hey, if there are no events the widget will be empty?

Comment (huocha) [reply src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/Stage.component.js:41]: hey @simonadomnisoru , yes based on the [mockup](https://www.sketch.com/s/95c3d586-218c-4685-968b-a388270da949/a/DvrbKa/play) the widget is empty

**Comment by simonadomnisoru** [comment]
Hey @huocha,

Congrats on this PR! Can you add some Cypress tests for the new functionalities added to the `StagesAndEventsWidget`? I think having some Scenarios testing how the DataTable behaves depending on how many events there are would be very helpful. WDYT?

Thank you!

**Comment by huocha** [comment]
> Hey @huocha,
> 
> Congrats on this PR! Can you add some Cypress tests for the new functionalities added to the `StagesAndEventsWidget`? I think having some Scenarios testing how the DataTable behaves depending on how many events there are would be very helpful. WDYT?
> 
> Thank you!

Thanks @simonadomnisoru for your kind comment. That's true it would be nice to have these tests, however most of the data currently only have 1 or 2 events and I can not create new event easily. I am working on finding a solution for testing. Thanks

**Pull request closed without merging**